superset-frontend/plugins/plugin-chart-cartodiagram/src/thematic/components/TimeSlider.tsx

```diff
@@ -55,30 +74,86 @@ export const TimeSlider: React.FC<TimeSliderProps> = props => {
     [timesliderTooltipFormat],
   );
 
+  const advanceSlider = useCallback(() => {
+    const currentIdx = markKeys.indexOf(currentValue);
+    const nextIdx = (currentIdx + 1) % markKeys.length;
+    const nextVal = markKeys[nextIdx];
+    setCurrentValue(nextVal);
+    onChange?.(nextVal);
+  }, [currentValue, markKeys, onChange]);
+
+  const togglePlay = () => setIsPlaying(prev => !prev);
+
   useEffect(() => {
-    if (!marks || !onChange) {
-      return;
+    if (isPlaying) {
+      intervalRef.current = setInterval(advanceSlider, 1000);
+    } else {
+      if (intervalRef.current) clearInterval(intervalRef.current);
```

Comment:
I'm not sure if the linter will fail due to one-line if-statements. Until we have a pipeline for that, it would be great if you could run

```bash
docker build --target lean --no-cache .
```

from project root?